From a8916e1ca7c2448ee5a5d31799b3300be697fe1a Mon Sep 17 00:00:00 2001
From: "penikelapati.kumar" <penikelapati.kumar@lge.com>
Date: Tue, 12 May 2020 13:44:43 +0530
Subject: [PATCH] Changes to support compilation

:Release Notes:
NMEAParser opensource library compilation support in WebOS.

:Detailed Notes:
Add CMakeLists.txt for compiling in webos.

:Testing Performed:
Build success

:QA Notes:
NA

:Issues Addressed:
N/A
---
Upstream-Status: Inappropriate [webOS specific]

 CMakeLists.txt                        |  7 +++++++
 Software/CMakeLists.txt               |  2 +-
 Software/NMEAParserLib/CMakeLists.txt | 15 ++++++++++-----
 files/pkgconfig/nmeaparser.pc         |  4 ++++
 4 files changed, 22 insertions(+), 6 deletions(-)
 create mode 100644 CMakeLists.txt
 create mode 100644 files/pkgconfig/nmeaparser.pc

diff --git a/CMakeLists.txt b/CMakeLists.txt
new file mode 100644
index 0000000..4241da4
--- /dev/null
+++ b/CMakeLists.txt
@@ -0,0 +1,6 @@
+#
+# CMakeLists - NMEAParserLib project
+#
+cmake_minimum_required(VERSION 2.8)
+add_subdirectory (Software)
+install(FILES "files/pkgconfig/nmeaparser.pc" DESTINATION "${CMAKE_INSTALL_DATAROOTDIR}/pkgconfig")
diff --git a/Software/CMakeLists.txt b/Software/CMakeLists.txt
index 2484ae0..afb24c4 100644
--- a/Software/CMakeLists.txt
+++ b/Software/CMakeLists.txt
@@ -5,5 +5,5 @@ cmake_minimum_required (VERSION 2.8)
 project (NMEAParserLib)
 
 add_subdirectory (NMEAParserLib)
-add_subdirectory (NMEAParserTest)
+#add_subdirectory (NMEAParserTest)
 
diff --git a/Software/NMEAParserLib/CMakeLists.txt b/Software/NMEAParserLib/CMakeLists.txt
index f653a90..985bf95 100644
--- a/Software/NMEAParserLib/CMakeLists.txt
+++ b/Software/NMEAParserLib/CMakeLists.txt
@@ -6,12 +6,15 @@ cmake_minimum_required(VERSION 2.8)
 project(NMEAParserLib)
 
 
-if(MSVC)
-	# Do nothing for now...
-else()
-        set(CMAKE_CXX_FLAGS " -Wall")
-endif ()
+#if(MSVC)
+#	# Do nothing for now...
+#else()
+#        set(CMAKE_CXX_FLAGS " -Wall")
+#endif ()
 
+#set(CMAKE_CXX_FLAGS " -Wall")
+set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wall")
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -std=c++11 -fPIC")
 FIND_PACKAGE(Git)
 
 execute_process(COMMAND git describe --abbrev=4 HEAD
@@ -46,6 +49,8 @@ add_library(NMEAParserLib
 	NMEASentenceRMC.h
 )
 
+install(TARGETS NMEAParserLib DESTINATION ${CMAKE_INSTALL_LIBDIR})
+install(DIRECTORY . DESTINATION ${CMAKE_INSTALL_PREFIX}/include/nmeaparser FILES_MATCHING PATTERN "*.h")
 #
 # Add additional libraries
 #
diff --git a/files/pkgconfig/nmeaparser.pc b/files/pkgconfig/nmeaparser.pc
new file mode 100644
index 0000000..7820987
--- /dev/null
+++ b/files/pkgconfig/nmeaparser.pc
@@ -0,0 +1,4 @@
+Name: nmeaparser
+Description: NMEA data parser for GPS data
+Version: 1.0.0
+Libs: -L/usr/lib -lNMEAParserLib
